Source/WebCore/html/parser/HTMLTreeBuilder.cpp

```diff
@@ -444,22 +437,23 @@ void HTMLTreeBuilder::processFakeCharacters(const String& characters)
 
 void HTMLTreeBuilder::processFakePEndTagIfPInButtonScope()
 {
-    if (!m_tree.openElements().inButtonScope(pTag->localName()))
+    if (!m_tree.openElements().inButtonScope(HTML::p))
         return;
-    AtomHTMLToken endP(HTMLToken::Type::EndTag, pTag->localName());
+    AtomHTMLToken endP(HTMLToken::Type::EndTag, TagName::p);
     processEndTag(WTFMove(endP));
 }
 
 namespace {
 
 bool isLi(const HTMLStackItem& item)
 {
-    return item.hasTagName(liTag);
+    return item.elementName() == HTML::li;
 }
 
 bool isDdOrDt(const HTMLStackItem& item)
 {
-    return item.hasTagName(ddTag) || item.hasTagName(dtTag);
+    return item.elementName() == HTML::dd
```

Comment:
I think putting them on a single line would read better when there are only a couple of items.